src/StackExchange.Redis/AzureMaintenanceEvent.cs

```diff
@@ -0,0 +1,143 @@
+                for (int i = 0; i < info?.Length / 2; i++)
+                {
+                    string key = null, value = null;
+                    if (2 * i < info.Length) { key = info[2 * i].Trim(); }
+                    if (2 * i + 1 < info.Length) { value = info[2 * i + 1].Trim(); }
+                    if (!string.IsNullOrEmpty(key) && !string.IsNullOrEmpty(value))
+                    {
+                        switch (key.ToLowerInvariant())
+                        {
+                            case "notificationtype":
+                                NotificationType = value;
+                                break;
+
+                            case "starttimeinutc":
+                                if (DateTime.TryParse(value, out DateTime startTime))
+                                {
+                                    StartTimeUtc = DateTime.SpecifyKind(startTime, DateTimeKind.Utc);
+                                }
+                                break;
+
+                            case "isreplica":
+                                bool.TryParse(value, out IsReplica);
+                                break;
+
+                            case "ipaddress":
+                                IPAddress.TryParse(value, out IpAddress);
+                                break;
+
+                            case "sslport":
+                                Int32.TryParse(value, out var port);
+                                SSLPort = port;
+                                break;
+
+                            case "nonsslport":
+                                Int32.TryParse(value, out var nonsslport);
+                                NonSSLPort = nonsslport;
+                                break;
+
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+            catch { }
+        }
+
+        internal async static Task AddListenerAsync(ConnectionMultiplexer multiplexer, LogProxy logProxy)
+        {
+            try
+            {
+                var sub = multiplexer.GetSubscriber();
+                if (sub == null)
+                {
+                    logProxy?.WriteLine("Failed to GetSubscriber for AzureRedisEvents");
+                    return;
+                }
+
+                await sub.SubscribeAsync("AzureRedisEvents", (channel, message) =>
+                {
+                    var newMessage = new AzureMaintenanceEvent(message);
+                    multiplexer.InvokeServerMaintenanceEvent(newMessage);
+
+                    if (StringComparer.OrdinalIgnoreCase.Equals(newMessage.NotificationType, "NodeMaintenanceEnded") || StringComparer.OrdinalIgnoreCase.Equals(newMessage.NotificationType, "NodeMaintenanceFailover"))
+                    {
+                        multiplexer.ReconfigureAsync(first: false, reconfigureAll: true, log: logProxy, blame: null, cause: "server maintenance").Wait();
```

Comment:
In the cause here, I recommend passing `$"Azure Event: {newMessage.NotificationType}" so we can tell which one it was in any debugger.